### ISSUE TITLE
Android Unable to instantiate activity

### ISSUE THREAD

**Issue by zorro2b** [NativeScript/NativeScript#3637]
### Did you verify this is a real problem by searching [Stack Overflow]
I have asked about it on SO:
http://stackoverflow.com/questions/42194688/nativescript-unable-to-instantiate-activity

### Tell us about the problem
I am trying the most basic steps of the hello world here: https://docs.nativescript.org/tutorial/chapter-1
Runs fine on iOS. Dies immediately on Android with exception shown in SO article.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
CLI: 2.5.0 cross platform modules: 2.5.0 plugins: 2.5.0

### Please tell us how to recreate the issue in as much detail as possible.
Here is the code I am using:
https://github.com/zorro2b/NativeScriptHelloWorld

tns run android
Device is running Android 5.0.2

**Comment by tsonevn** [comment]
Hi @zorro2b,
I tested your project on my side and it seems to work as expected.
![screencast 2017-02-15 at 8 33 54 am](https://cloud.githubusercontent.com/assets/17448734/22963257/be18ab6a-f35a-11e6-8f8c-9da7d14210af.gif)
You could try also to clear npm cache with `npm cache clear` as well as to delete `platforms`, `node_modules` and `hooks` folders. To be sure that the app will be deployed correctly, you also can remove the app from the simulator or device. Then build your project again with `tns run android`

In case this does not help you, please build the project with `tns run android --log trace` and send us the full log from the console.

**Comment by zorro2b** [comment]
Still fails. Log attached.
[typescript.txt.zip](https://github.com/NativeScript/NativeScript/files/776380/typescript.txt.zip)

**Comment by tsonevn** [comment]
Hi @zorro2b,
Thank you for sending us the log.
However, we will need some more info about the problem. Could you run `tns doctor` command, which will check whether there is an issue with environment setup? 
You could also try to deploy the app on an emulator with `tns run android --emulator` and to verify whether you will have the same issue. If the app has been deployed succesfully, please provide some more info about the device.
You can also build the project with `tns run android` and to send us again the log from the console.

**Comment by zorro2b** [comment]
az-mbp-ether:NativeScriptHelloWorld andrew$ tns doctor
Verifying CocoaPods. This may take more than a minute, please be patient.
  ◡ Installing iOS runtime.nativescript-check-cocoapods@0.0.1 /private/var/folders/mr/9kgxg_v15sx5vsrsdlyv9s7m0000gn/T/nativescript-check-cocoapods117115-46617-1iueka8.agntrzfr
└── tns-ios@2.5.0


  ◟ Verifying CocoaPods. This may take some time, please be patient..
Your components are up-to-date.

No issues were detected.

**Comment by zorro2b** [comment]
Also fails on emulator Nexus_6_API24
[typescript.txt.zip](https://github.com/NativeScript/NativeScript/files/776649/typescript.txt.zip)

Emulator details:


Name: Nexus_6_API_24

CPU/ABI: Google APIs Intel Atom (x86_64)

Path: /Users/andrew/.android/avd/Nexus_6_API_24.avd

Target: google_apis [Google APIs] (API level 24)

Skin: nexus_6

SD Card: 100M

hw.dPad: no

runtime.network.speed: full

hw.accelerometer: yes

hw.device.name: Nexus 6

vm.heapSize: 64

skin.dynamic: yes

hw.device.manufacturer: Google

hw.gps: yes

hw.initialOrientation: Portrait

image.androidVersion.api: 24

hw.audioInput: yes

image.sysdir.1: system-images/android-24/google_apis/x86_64/

tag.id: google_apis

showDeviceFrame: yes

hw.camera.back: emulated

hw.mainKeys: no

AvdId: Nexus_6_API_24

hw.camera.front: emulated

hw.lcd.density: 560

avd.ini.displayname: Nexus 6 API 24

hw.gpu.mode: auto

hw.device.hash2: MD5:9b564b60b1aebee32c73ded9daa1e620

hw.ramSize: 1536

hw.trackBall: no

hw.battery: yes

hw.cpu.ncore: 4

hw.sdCard: yes

tag.display: Google APIs

runtime.network.latency: none

hw.keyboard: yes

hw.sensors.proximity: yes

disk.dataPartition.size: 800M

hw.sensors.orientation: yes

avd.ini.encoding: UTF-8

hw.gpu.enabled: yes

**Comment by tsonevn** [comment]
Hi @zorro2b,
We verified your problem on our side on an emulator with the same setup, however, we are still unable to reproduce this behavior. 
To be able to investigate further, what is causing this issue, you could follow the below-given steps.
1. open new tap on the terminal.
2. run `adb logcat --clear`, which will clear selected buffers.
3. connect the device to the mac.
4. run `adb logcat` in terminal 
5. start the app on the device, when the application crashed, stop `adb logcat` and send us again the log from the tab. 

Regards,
@tsonevn

**Comment by zorro2b** [comment]
Please see attached.
[typescript.zip](https://github.com/NativeScript/NativeScript/files/778089/typescript.zip)

**Comment by tsonevn** [comment]
Hi @zorro2b ,
Thank you for your cooperation.

I was able to reproduce this behavior while deploying an app on a device connected via WiFi. `tns run android` will start livesync automatically and this case some of the files will not be live synced correctly. If this is the case, could you verify, whether you will have the same behavior while using a device connected via USB? 
Before building the app again you could delete again `platforms`, `node_modules` and `hooks` folders.

Let me know whether this helps.

**Comment by zorro2b** [comment]
Hey @tsonevn,
I have been doing all my device testing via USB. I also had the same result with an emulator. Were you able to reproduce with an emulator?

**Comment by tsonevn** [comment]
Hi @zorro2b,

At the moment we are able to reproduce this behavior only on a real device connected via WiFi. Your project has been built correctly on our side on an emulator.
Could you send us again the full project with platforms and node_modules folders?

I look forward to hearing from you and thank you again for your cooperation.

**Comment by basitjawaidgranjur** [comment]
@tsonevn Hello,
I also encounter the same issue, tested on ios and it seems to work fine, but for Android the build failed.

**Comment by zorro2b** [comment]
@tsonevn here is the whole project dir.
https://dl.dropboxusercontent.com/u/291335/NativeScriptHelloWorld.zip

**Comment by tsonevn** [comment]
Hi @zorro2b, 

Thank you for the project. 
We will need some time to investigate this behavior and to find what is causing this issue.
You could keep track on this issue and we will provide more info about the problem and how it could be fixed.

Regards,
@tsonevn

**Comment by Pip3r4o** [comment]
Hey @zorro2b 

From the logs and the APK that you've so kindly provided I was able to identify that the problem originates from the fact that one of the build steps doesn't execute at all, and thus the application package ends up incomplete. 

I was however unable to reproduce the same behavior on my end on Windows OS.

Please remove the installed platforms, and then build for android with logging enabled `--log trace`. Meanwhile we'll continue investigating the possible causes for that strange behavior. Thanks!

**Comment by zorro2b** [comment]
So it works now!
What I did was upgrade to nativescript 2.5.1. I also noticed that there was an error during the upgrade about access to /Users/andrew/.local/share/.nativescript-cli/user-settings.json so I gave that write permission. Removed platforms, node_modules and hooks dirs, did the rebuild and it worked!
Attached is the log anyway in case it is helpful.
[typescript.zip](https://github.com/NativeScript/NativeScript/files/788395/typescript.zip)
